### PR TITLE
Dasc 1208/add labels to bq job config

### PR DESCRIPTION
**Pull request by chagasVinicius**
#### Please provide details about this Pull Request (why the change is being made, what this commit will do):
The PR adds labels to the BigQuery.JobConfig in order to help us understand the job's performances better and enable new segmentation strategies for the Carol Apps.

#### Please provide links to relevant Trello cards, Slab topics or support tickets:
https://totvslabs.atlassian.net/browse/DASC-1208?atlOrigin=eyJpIjoiOWZhZTc4ZDViYmUxNDhlNmIyNWVkOGY4YWRmNTc0ZjciLCJwIjoiaiJ9

#### QA
image

#### Attention Points
- ~Although the behaviour of the `BQ.query` function has not changed in a production test. I was unable to verify that the labels were added to the BigQuery job because I don't have access to the tables. Maybe there is another way to check this in production, but I couldn't find it.~
- ~I added some doubts in the code to check if my decision was correct for the respective labels.~

### CODE REVIEW

File: pycarol/bigquery.py [hunk 1]
```diff
@@ -254,6 +255,31 @@ def _generate_client(service_account: T.Dict) -> bigquery.Client:
...
+        label_keys = [
+            "tenant_id",
+            "tenant_name",
+            "organization_id",
+            "organization_name",
+            "job_type",  ## Is any case that using this class is async?
+            "source",
+            "task_id",  ## In this case may we use the "mdmBigQueryProvisionId" from 'carol._current_env()'?
+            "carol_app_name",
+            "carol_app_process_name",  ## I do not understand what should be this property
```
Comment (darolt) [inline pycarol/bigquery.py:279]: image
This property points to the Carol App process. Batch apps may have many processes while online apps usually have a single process. Operator sets environment variables before starting the processes. The variable you are looking for is ALGORITHM_NAME (see the image). You may use os package to get this value.

File: pycarol/bigquery.py [hunk 2]
```diff
@@ -254,6 +255,31 @@ def _generate_client(service_account: T.Dict) -> bigquery.Client:
...
+            "",
+            self._env.get("app_name", ""),
+            "",
+        ]
+        label_keys = [
+            "tenant_id",
+            "tenant_name",
+            "organization_id",
+            "organization_name",
+            "job_type",  ## Is any case that using this class is async?
```
Comment (darolt) [inline pycarol/bigquery.py:275]: I think that the way we are querying BQ is always sync.

File: pycarol/bigquery.py [hunk 3]
```diff
@@ -254,6 +255,31 @@ def _generate_client(service_account: T.Dict) -> bigquery.Client:
...
+            "source",
+            "task_id",  ## In this case may we use the "mdmBigQueryProvisionId" from 'carol._current_env()'?
+            "carol_app_name",
+            "carol_app_process_name",  ## I do not understand what should be this property
+        ]
+        return {k: v for k, v in zip(label_keys, label_values) if v.strip() != ""}
```
Comment (darolt) [inline pycarol/bigquery.py:281]: why not creating the dict directly instead of creating keys and values and zipping them?

**Review by darolt — CHANGES_REQUESTED** [review]

File: pycarol/bigquery.py (hunk 3 above)
Comment (chagasVinicius) [reply pycarol/bigquery.py:281]: Do you mean like this?
```python
labels = {
  "tenant_name" : "tenant_name",
  "carol_app_name" : "",
  ...
}
return {k:v for k,v in labels.items() if v.strip() != ""}
``` 
Sure, makes sense.

Comment (darolt) [reply pycarol/bigquery.py:281]: yep.

**Review by darolt — APPROVED** [review]
LGTM

**Pull request merged**